beacon-chain/p2p/service.go

```diff
@@ -105,7 +105,13 @@ func (s *Service) Status() error {
 
 // Encoding returns the configured networking encoding.
 func (s *Service) Encoding() encoder.NetworkEncoding {
-	// TODO(3147): Return based on flag value
+	encoding := s.cfg.Encoding
+	switch encoding {
```

Comment:
Please add default to panic if invalid flag.

Reply:
alright,done